src/conduits/RwaSwapInputConduit2.sol

```diff
@@ -0,0 +1,329 @@
+     * @param amt The amount of `token` flushed out.
+     */
+    event Yank(address indexed token, address indexed usr, uint256 amt);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaSwapInputConduit2/not-authorized");
+        _;
+    }
+
+    modifier onlyMate() {
+        require(may[msg.sender] == 1, "RwaSwapInputConduit2/not-mate");
+        _;
+    }
+
+    /**
+     * @notice Defines addresses and gives `msg.sender` admin access.
+     * @param _psm PSM contract address.
+     * @param _dai DAI contract address.
+     * @param _gem GEM contract address.
+     * @param _to RwaUrn contract address.
+     */
+    constructor(
+        address _dai,
+        address _gem,
+        address _psm,
+        address _to
+    ) public {
+        require(_to != address(0), "RwaSwapInputConduit2/invalid-to-address");
+        require(PsmAbstract(_psm).dai() == _dai, "RwaSwapInputConduit2/wrong-dai-for-psm");
+        require(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem() == _gem, "RwaSwapInputConduit2/wrong-gem-for-psm");
+
+        // We assume that DAI will alway have 18 decimals
+        to18ConversionFactor = 10**_sub(18, GemAbstract(_gem).decimals());
+
+        psm = PsmAbstract(_psm);
+        dai = DaiAbstract(_dai);
+        gem = GemAbstract(_gem);
+
+        to = _to;
+
+        // Give unlimited approval to PSM gemjoin
+        GemAbstract(_gem).approve(address(psm.gemJoin()), type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"to", "quitTo", "psm"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            quitTo = data;
```

Comment:
Consider checking `quitTo != address(0)` here.

Reply:
We already have the check on `_doQuit`. Actually we might want to set it to `address(0)` to effectively disable `quit()`.

Reply:
Yeah that was why I brought up checking it here. But I guess it is reasonable that you might want to quickly revoke all ability to `quit` instead of disabling each whitelisted address individually.